src/components/LayoutTree.tsx

```diff
@@ -330,6 +331,7 @@ const linearizeTree = (
       simplePath: contextViewActive ? thoughtToPath(state, child.id) : appendToPathMemo(simplePath, child.id),
       style,
       thoughtId: child.id,
+      value: child.value,
```

Comment:
Rather than pass `value: child.value`, pass `isEmpty: child.value === ''` so that `treeThoughts` is not recalculated every time a thought value changes.

In general, prefer narrower API's that pass the minimum data needed by the downstream components. This will decrease the surface area for change and in React often increase performance.

Reply:
Thanks for the suggestion. I've updated the logic so that it passes `isEmpty` instead of `value`.